### PR TITLE
scripts for using openstack cluster

### PR DESCRIPTION
**Pull request by tomsilver**
because supercloud is continuing to be difficult!!!!

see the README and file docstrings for info.

### CODE REVIEW

**Review by wester-is-not-real — APPROVED** [review]
Looks good!

File: scripts/openstack/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Instructions for Using Openstack
+
+1. Request an account through the LIS group. See [here](https://tig.csail.mit.edu/shared-computing/open-stack/quick-start/).
+2. Follow the instructions for creating and uploading a private key.
+3. Launch instances using the `predicators` image with your private key. Make sure you launch enough instances so that each (environment, approach, seed) can have one instance.
+4. Create a file (e.g. `machines.txt`) that lists your instance IP addresses, one per line.
+5. Create an experiment yaml file (see `scripts/openstack/configs/example.yaml` for an example).
+6. Run `python scripts/openstack/launch.py --config <config file> --machines <machines file>--sshkey <private key file>` to launch your experiments.
```
Comment (ronuchit) [inline scripts/openstack/README.md:8]: should there be a space before "--sshkey"?

File: scripts/openstack/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Instructions for Using Openstack
+
+1. Request an account through the LIS group. See [here](https://tig.csail.mit.edu/shared-computing/open-stack/quick-start/).
+2. Follow the instructions for creating and uploading a private key.
+3. Launch instances using the `predicators` image with your private key. Make sure you launch enough instances so that each (environment, approach, seed) can have one instance.
+4. Create a file (e.g. `machines.txt`) that lists your instance IP addresses, one per line.
```
Comment (ronuchit) [inline scripts/openstack/README.md:6]: is there a script for doing this?

File: scripts/openstack/launch.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+Requires a file that contains a list of IP addresses for instances that are:
+    - Turned on
+    - Accessible via ssh for the user of this file
+    - Configured with a predicators image
+    - Sufficient in number to run all of the experiments in the config file
+
+Usage example:
+    python scripts/openstack/launch.py --config example.yaml \
+        --machines machines.txt --sshkey ~/.ssh/cloud.key
```
Comment (ronuchit) [inline scripts/openstack/launch.py:11]: why can't machines.txt be autogenerated from the config?

File: scripts/openstack/launch.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+
+def _launch_experiment(cmd: str, machine: str, logfile: str,
+                       ssh_key: str) -> None:
+    print(f"Launching on machine {machine}: {cmd}")
+    server_cmds = [
+        # Prepare the predicators directory.
+        "cd ~/predicators",
+        "mkdir -p logs",
+        "git checkout master",
```
Comment (ronuchit) [inline scripts/openstack/launch.py:108]: is it awkward that you can only run master? this could be part of the config instead right?

**Review by ronuchit — APPROVED** [review]
smh supercloud

File: scripts/openstack/launch.py (hunk 3 above)
Comment (tomsilver) [reply scripts/openstack/launch.py:11]: you can have persistent machines -- I have 30 static IP addresses right now on my account for example. the thought is that you would just make machines.txt once, or infrequently at least, and then run many different configs

**Pull request merged**